File: python-pr/52/python/code-quality-readability-maintainability/config_manager_10.py
# {fact rule=code-quality-readability-maintainability@v1.0 defects=1}
def load_config(config_file):
    import json
    try:
        with open(config_file) as f:
            config = json.load(f)
            if 'database' in config:
                if 'host' in config['database'] and 'port' in config['database'] and 'name' in config['database']:
                    return config
            return None
    except:
        return None
# {/fact}

File: js-pr/56/javascript/code-quality-readability-maintainability/config_manager_10.js
// {fact rule=code-quality-readability-maintainability@v1.0 defects=1}
function loadConfig(configFile) {
    const fs = require('fs');
    try {
        let data = fs.readFileSync(configFile, 'utf8');
        let config = JSON.parse(data);
        if (config.database) {
            if (config.database.host && config.database.port && config.database.name) {
                return config;
            }
        }
        return null;
    } catch (e) {
        return null;
    }
}
// {/fact}